### PR TITLE
Compute curvature on mesh boundary and use in model specifications

### PR DESCRIPTION
**Pull request by emmetfrancis**
This PR includes the following additions to SMART:

In `mesh_tools`:
- Add function `compute_curvature`, which computes the mean curvature over specified boundaries and writes out curvatures as a vertex mesh function over the parent mesh. Currently only tested in 2D.
- Add `return_curvature` as an argument to `create_2dcell`. If only a half cell geometry is desired, curvatures are computed over the whole cell contour to avoid errors at the x = 0 axis, then linearly interpolated onto the half-cell mesh.

In `mesh`:
- Add `curvature` as an attribute of `parent_mesh` objects (is empty by default).

In `model`:
- If `parent_mesh.curvature` is not empty, each compartment has an associated dolfin function `curv_func` containing curvatures from the vertex mesh function mapped onto the compartment's FunctionSpace using the new function `model.mf0_to_fun`.
- If user defines an initial condition that contains `curv` as a free symbol, then the associated `curv_func` is substituted into this expression when evaluating the initial condition in `model.dolfin_set_function_values`

In `model_assembly`:
- Add `curv` to equation variables for reaction that include `curv` as a free symbol.

A new version of example 2, `example2_withcurv` has been added to demonstrate use of the new features.

### CODE REVIEW

**Comment by emmetfrancis** [comment]
To do before merging:
- Test curvature calculations in 3D.
- Alter curvature definition for axisymmetric geometries defined on 2D mesh.
- Consider defining curvature as a parameter (would require implementing spatially-dependent parameters, maybe in a separate PR)

**Comment by emmetfrancis** [comment]
First two points above have been completed (curvature needed to be divided by 2 for consistency in 3D). Spatially-dependent parameters may be implemented in a future PR.

File: smart/mesh_tools.py [hunk 1]
```diff
@@ -812,6 +826,161 @@ def meshSizeCallback(dim, tag, x, y, z, lc):
...
+        ValueError("Each listed facet marker value must have an associated cell marker value")
+
+    if half_mesh_data == "":
+        kappa_mf = d.MeshFunction("double", ref_mesh, 0)
+    else:
+        (dmesh_half, mf2_half) = half_mesh_data
+        kappa_mf = d.MeshFunction("double", dmesh_half, 0)
+
+    for i in range(len(cell_marker_vec)):
+        mesh = d.MeshView.create(mf_cell, cell_marker_vec[i])
```
Comment (jorgensd) [inline smart/mesh_tools.py:875]: Note for later: we could use the new `create_meshview` function in dolfin to avoid some, But not all parallel issued

**Comment by emmetfrancis** [comment]
Added the few curvature calculation tests to the "tests" folder. The calculations in 3D are quite inaccurate here (likely very sensitive to the discretization); we could look into improving this calculation in the future.

**Review by finsberg — APPROVED** [review]

**Pull request merged**